main/main.cpp

```diff
@@ -323,147 +333,142 @@ void finalize_theme_db() {
+		String display_driver = DisplayServer::get_create_function_name(i);
+		String rendering_drivers = "'" + String("', '").join(DisplayServer::get_create_function_rendering_drivers(i)) + "'";
+		drivers.push_back("'" + display_driver + "' (" + rendering_drivers + ")");
+	}
+	return String(", ").join(drivers);
+}
+
+// Use the --help command line argument to show all the options Godot supports.
+void Main::print_help(const char *p_binary) {
+	print_line(VERSION_NAME + " v" + get_full_version_string() + " - " + VERSION_WEBSITE);
+	print_line("Free and open source software under the terms of the MIT license.");
+	print_line("(c) 2014-present Godot Engine contributors.");
+	print_line("(c) 2007-2014 Juan Linietsky, Ariel Manzur.");
+	print_line();
+
+	print_line("Usage: " + String(p_binary) + " [options] [path to scene or 'project.godot' file]");
+	print_line();
+
+	print_line("General options:");
+	print_line("  -h, --help                        Display this help message.");
+	print_line("  --version                         Display the version string.");
+	print_line("  -v, --verbose                     Use verbose stdout mode.");
+	print_line("  -q, --quiet                       Quiet mode, silences stdout messages. Errors are still displayed.");
+	print_line();
+
+	print_line("Run options:");
+	print_line("  --, ++                            Separator for user-provided arguments. Following arguments are not used by the engine, but can be read from `OS.get_cmdline_user_args()`.");
+#ifdef TOOLS_ENABLED
+	print_line("  -e, --editor                      Start the editor instead of running the scene.");
+	print_line("  -p, --project-manager             Start the project manager, even if a project is auto-detected.");
+	print_line("  --debug-server <uri>              Start the editor debug server (<protocol>://<host/IP>[:<port>], e.g. tcp://127.0.0.1:6007)");
+#endif
+	print_line("  --quit                            Quit after the first iteration.");
+	print_line("  -l, --language <locale>           Use a specific locale (<locale> being a two-letter code).");
+	print_line("  --path <directory>                Path to a project (<directory> must contain a 'project.godot' file).");
+	print_line("  -u, --upwards                     Scan folders upwards for project.godot file.");
+	print_line("  --main-pack <file>                Path to a pack (.pck) file to load.");
+	print_line("  --render-thread <mode>            Render thread mode ['unsafe', 'safe', 'separate'].");
+	print_line("  --remote-fs <address>             Remote filesystem (<host/IP>[:<port>] address).");
+	print_line("  --remote-fs-password <password>   Password for remote filesystem.");
+	print_line("  --audio-driver <driver>           Audio driver [" + get_audio_drivers() + "].");
+	print_line("  --display-driver <driver>         Display driver (and rendering driver) [" + get_display_drivers() + "].");
```

Comment:
I extracted these two blocks as helper functions since the nested for loops made it harder to quickly skim `print_help()`.